force-app/main/kontaktskjemaKafka/classes/KafkaKontaktSkjemaHandler_Helper.cls

```diff
@@ -0,0 +1,134 @@
+public with sharing class KafkaKontaktSkjemaHandler_Helper {
+    public static Id inclusionRecordType = Schema.SObjectType.CustomOpportunity__c.getRecordTypeInfosByDeveloperName()
+    .get('Inclusion')
+    .getRecordTypeId();
+
+    
+    public static List<KafkaKontaktSkjemaHandler_Model> convertKafkaToOpportunity(List<KafkaMessage__c> messages) {
+        List<KafkaKontaktSkjemaHandler_Model> kontaktskjema = new List<KafkaKontaktSkjemaHandler_Model>();
+        for (KafkaMessage__c msg : messages) {
+            try {
+                msg.CRM_Status__c = 'Processed';
+                String decodedValue = KafkaMessageUtils.base64ULSafeDecode(msg.CRM_Value__c).toString();
+                KafkaKontaktSkjemaHandler_Model o = KafkaKontaktSkjemaHandler_Model.parse(decodedValue);
+                kontaktskjema.add(o);
+            } catch (Exception e) {
+                msg.CRM_ErrorMessage__c = e.getMessage() + ' (' + e.getLineNumber() + ')';
+            }
+        }
+        return kontaktskjema;
+    }
+
+    public static List<CustomOpportunity__c> convertKontaktskjemaToOpportunity(
+        List<KafkaKontaktSkjemaHandler_Model> kontaktskjema
+    ) {
+        List<CustomOpportunity__c> opportunity = new List<CustomOpportunity__c>();
+        for (KafkaKontaktSkjemaHandler_Model k : kontaktskjema) {
+            Id contactId = fetchOrCreateContact(k);
```

Comment:
Dette kan være problematisk ved behandling av store datamengder pga SOQL limits. `fetchOrCreateContact()` bruker `SharedFunctions.fetchAccounts()` som gjør en SOQL spørring for å hente accounts. SharedFunctions.fetchOrCreateContact() bruker `SharedFunctions_FetchOrCreateContact.getExistingContacts()` som gjøre en SOQL for å hente kontaktene. 

Dette betyr at flere SOQL spørringer gjøres for hver interasjon av for loopen. Dette bør være utenfor loopen.